runtime/runtime-params-estimator/src/gas_cost.rs

```diff
@@ -124,13 +178,29 @@ impl PartialOrd for GasCost {
 
 impl Ord for GasCost {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.value.cmp(&other.value)
+        self.scalar_cost().cmp(&other.scalar_cost())
     }
 }
 
+// See runtime/runtime-params-estimator/emu-cost/README.md for the motivation of constant values.
+const READ_BYTE_COST: u64 = 27;
+const WRITE_BYTE_COST: u64 = 47;
+
 impl GasCost {
     pub(crate) fn to_gas(&self) -> Gas {
-        ratio_to_gas(self.metric, self.value)
+        ratio_to_gas(self.metric, self.scalar_cost())
+    }
+
+    /// Reduces the detailed measurement to a single number.
+    pub(crate) fn scalar_cost(&self) -> Ratio<u64> {
```

Comment:
I am somewhat surprised that we need this method, and for it to be public, I'd expect that `to_gas` would be enough.

Reply:
Yeah this is a bit ugly right now. It's mostly to keep refactoring simple.

There are places where the code uses a single raw `u64` before applying the ratios, instead of `GasCost` or simply `Gas`. For example the `least_squares_method` in `runtime-params-estimator/src/estimator_context.rs` operates on such values and the conversion to gas happens afterwards.

I am pretty sure it could be rewritten to use `Gas` in all those places. It would also nicely lead to the removal of all calls to `ratio_to_gas` and `ratio_to_gas_signed`, as far as I can tell.

I will need to touch quite a bit more code but I think it is worth the effort.